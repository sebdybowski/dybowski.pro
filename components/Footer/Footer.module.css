.Footer {
  height: 20rem;
  display: flex;
  z-index: 0;
  align-items: flex-end;
}

.Footer > .Content {
  z-index: 1024;
  position: absolute;
  border-top: 2px solid var(--text-color);
  margin: 0 6.5rem;
  width: calc(100% - 13rem);
  display: flex;
  flex-direction: column;
  background-color: var(--background-color);
  padding: 2rem 0;
  align-items: center;
}