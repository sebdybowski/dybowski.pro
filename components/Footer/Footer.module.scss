@use 'styles/variables.scss' as *;

.Footer {
  height: 20vh;
  display: flex;
  background-color: $brand-color-primary;
  background-image: $brand-bg;
  flex-direction: column;
  justify-content: center;
}

.Footer>.Content {
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  color: $text-color-light;
}

.Footer>.Content>.Year {
  font-size: 2rem;
}

.Footer>.Content>.Year>.Year__End {
  font-size: 3rem;
  font-weight: bold;
}

.Footer>.Content>.Caption {
  font-size: 1rem;
  line-height: 2rem;
}

.Footer>.Content>.Caption a {
  text-decoration: underline;
}