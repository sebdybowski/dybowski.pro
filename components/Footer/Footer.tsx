import React from 'react';
import styles from './Footer.module.css';
import Image from 'next/image';

const Footer = () => (
  <footer className={styles.Footer} data-testid="Footer">
    <div className={styles.Line} />
    {/* <Image
      src="/footer-background.svg"
      alt="footer background"
      layout='fill'
      objectPosition='bottom'
    /> */}
    <div className={styles.Content}>
      <span>2022</span>
      <span>Built with NEXT.JS</span>
    </div>
    <div className={styles.Line} />
  </footer>
);

export default Footer;
