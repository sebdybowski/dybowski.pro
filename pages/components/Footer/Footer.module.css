.Footer {
  height: 20rem;
  display: flex;
  background-image: url('/footer-background.svg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: top;
  z-index: 0;
  align-items: flex-end ;
}

.Footer > .Content {
  z-index: 1024;
  position: absolute;
  border-top: 2px solid var(--text-color);
  margin: 0 6.5rem;
  width: calc(100% - 13rem);
  display: flex;
  flex-direction: column;
  background-color: var(--background-color);
  padding: 2rem 0;
  align-items: center
};