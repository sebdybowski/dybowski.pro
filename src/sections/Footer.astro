---
import { Github, Linkedin, Twitter, Mail } from "lucide-preact";

const currentYear = new Date().getFullYear();
const links = [
  { url: "https://github.com/sebdybowski", label: "Github", Icon: Github },
  {
    url: "https://pl.linkedin.com/in/sebastian-dybowski",
    label: "LinkedIn",
    Icon: Linkedin,
  },
  {
    url: "https://twitter.com/sebdybowskidev",
    label: "Twitter",
    Icon: Twitter,
  },
  {
    url: "mailto:dybowski.seb@gmail.com",
    label: "E-mail",
    Icon: Mail,
  },
];
---

<footer class="container-hybrid">
  <div class="grid">
    <div>
      <h4>{currentYear} / Seb Dybowski</h4>
      <hr />
      <p>
        Proudly created with
        <a href="https://astro.build/" target="_blank">Astro</a> and <a
          href="https://picocss.com/"
          target="_blank">Pico.css</a
        >.
      </p>
      <p>
        Advocating for <i>Free and Open Source Software</i>.<br /> Check out this
        website's
        <a href="https://github.com/sebdybowski/dybowski.pro">source code</a>.
      </p>
    </div>
    <div>
      <h4>Connect</h4>
      <hr />
      <ul>
        {
          links.map(({ url, label, Icon }) => (
            <li>
              <a href={url} target="_blank">
                <Icon size="1rem" class="icon" /> / {label}
              </a>
            </li>
          ))
        }
      </ul>
    </div>
  </div>
</footer>

<style>
  footer {
    background: var(--card-background-color);
    box-shadow: inset var(--card-box-shadow);
  }

  footer > div {
    padding: var(--block-spacing-vertical) 0;
  }

  hr {
    margin-bottom: 2rem;
  }

  .icon {
    margin-top: -0.25rem;
  }
</style>
