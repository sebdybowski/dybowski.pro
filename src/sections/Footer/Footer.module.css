.footer {
  background: var(--card-background-color);
  box-shadow: inset var(--card-box-shadow);
  display: flex;
  flex-direction: row;
  padding-top: 2rem;
  padding-bottom: 2rem;

  @media screen and (max-width: 992px) {
    flex-direction: column;
  }
}

.footer__item {
  padding: var(--block-spacing-vertical) 0;
  flex: 1;
}

.divider {
  margin-bottom: 2rem;
}

.icon {
  margin-top: -0.25rem;
}