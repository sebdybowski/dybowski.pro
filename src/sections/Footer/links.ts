const connect = [
    {
        url: "https://github.com/sebdybowski",
        label: "Github",
        icon: "lucide:github",
    },
    {
        url: "https://pl.linkedin.com/in/sebastian-dybowski",
        label: "LinkedIn",
        icon: "lucide:linkedin",
    },
    {
        url: "https://twitter.com/sebdybowskidev",
        label: "Twitter (X)",
        icon: "lucide:twitter",
    },
];

const contact = [
    {
        label: "Mobile",
        value: "+48 782 710 459",
        url: "tel:+48782710459",
        icon: "lucide:phone",
    },
    {
        label: "E-mail",
        value: "dybowski.seb@gmail.com",
        url: "mailto:dybowski.seb@gmail.com",
        icon: "lucide:mail",
    },
];

export { connect, contact };

