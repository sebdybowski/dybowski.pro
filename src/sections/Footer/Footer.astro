---
import { Icon } from "astro-icon/components";
import styles from "./Footer.module.css";
import { connect, contact } from "./links";

const currentYear = new Date().getFullYear();
---

<footer class={styles.footer}>
  <div class="container-hybrid grid">
    <div class={styles.footer__item}>
      <h3>{currentYear} / Seb Dybowski</h3>
      <hr class={styles.divider} />
      <p>
        Proudly created with
        <a href="https://astro.build/" target="_blank">Astro</a> and <a
          href="https://picocss.com/"
          target="_blank">Pico.css</a
        >.
      </p>
      <p>
        Advocating for <i>Free and Open Source Software</i>.<br />
        Check out this website's
        <a href="https://github.com/sebdybowski/dybowski.pro"> source code</a>.
      </p>
    </div>
    <div class={styles.footer__item}>
      <h3>Contact</h3>
      <hr class={styles.divider} />
      <ul>
        {
          contact.map(({ label, value, icon, url }) => (
            <li>
              <Icon name={icon} />
              {label}:{" "}
              <a href={url} target="_blank">
                {value}
              </a>
            </li>
          ))
        }
      </ul>
    </div>
    <div class={styles.footer__item}>
      <h3>Connect</h3>
      <hr class={styles.divider} />
      <ul>
        {
          connect.map(({ url, label, icon }) => (
            <li>
              <a href={url} target="_blank">
                <Icon name={icon} /> / {label}
              </a>
            </li>
          ))
        }
      </ul>
    </div>
  </div>
</footer>
