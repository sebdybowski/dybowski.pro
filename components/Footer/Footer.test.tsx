import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import Footer from './Footer';

describe('<Footer />', () => {
  test('it should mount', () => {
    const { asFragment } = render(<Footer />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
