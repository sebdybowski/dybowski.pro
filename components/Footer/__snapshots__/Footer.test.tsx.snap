// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> it should mount 1`] = `
<DocumentFragment>
  <footer
    class="Footer"
    data-testid="Footer"
  >
    <div
      class="Content"
    >
      <span
        class="Year"
      >
        20
        <span
          class="Year__End"
        >
          22
        </span>
      </span>
      <span
        class="Caption"
      >
        Built with 
        <a
          href="https://nextjs.org/"
          rel="noreferrer"
          target="_blank"
        >
          Next.js
        </a>
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
